libsolidity/ast/Types.cpp

```diff
@@ -453,8 +456,13 @@ TypePointer FixedPointType::binaryOperatorResult(Token::Value _operator, TypePoi
 	if (Token::isBitOp(_operator) || Token::isBooleanOp(_operator))
 		return TypePointer();
 	if (auto fixType = dynamic_pointer_cast<FixedPointType const>(commonType))
+	{
 		if (Token::Exp == _operator)
 			return TypePointer();
+	}
+	else if (auto intType = dynamic_pointer_cast<IntegerType const>(commonType))
+		if (intType->isAddress())
```

Comment:
I don't think this can happen (because `Type::commonType` will not return an address)